cmd/influx/importer/v8.go

```diff
@@ -0,0 +1,222 @@
+package importer
+
+import (
+	"bufio"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/influxdb/influxdb/client"
+)
+
+const batchSize = 5000
+
+// V8Config is the config used to initialize a V8 importer
+type V8Config struct {
+	username, password string
+	url                url.URL
+	precision          string
+	writeConsistency   string
+	file, version      string
+	compressed         bool
+}
+
+// NewV8Config returns an initialized *V8Config
+func NewV8Config(username, password, precision, writeConsistency, file, version string, u url.URL, compressed bool) *V8Config {
+	return &V8Config{
+		username:         username,
+		password:         password,
+		precision:        precision,
+		writeConsistency: writeConsistency,
+		file:             file,
+		version:          version,
+		url:              u,
+		compressed:       compressed,
+	}
+}
```

Comment:
If the fields are exported then you can drop the argument list and just return a config with default settings.